### PR TITLE
Fallback to corrupt element when encountering unknown elem type

### LINKED ISSUE #18576: Park from #8340 can no longer be opened
### Operating System

Any

### OpenRCT2 build

0.4.0

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

_No response_

### Describe the issue

Park attached to issue #8340 can no longer be loaded correctly ever since v0.4.0. It used to load without any issues in v0.3.5.1, the previous release.

https://github.com/OpenRCT2/OpenRCT2/files/2647348/park_performance_8340.sv6.gz

### Steps to reproduce

Try loading the park, observe failure and `This tile element type cannot be converted directly!` getting printed out in logs

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by janisozaur**
Fixes #18576

### CODE REVIEW

**Comment by janisozaur** [comment]
https://github.com/OpenRCT2/OpenRCT2/pull/18577#issuecomment-1313110443

Additionally: should any effort be made to see if there are consecutive corrupt elements and should they get merged to single one?

**Comment by Gymnasiast** [comment]
I’m going to investigate the file, please do not merge until I approve.

**Review by Gymnasiast — DISMISSED** [review]
This looks correct. This just needs a changelog entry, then.

File: src/openrct2/rct12/RCT12.cpp [hunk 1]
```diff
@@ -30,7 +30,25 @@ using namespace OpenRCT2;
...
+        case RCT12TileElementType::Wall:
+        case RCT12TileElementType::LargeScenery:
+        case RCT12TileElementType::Banner:
+        case RCT12TileElementType::Corrupt:
+        case RCT12TileElementType::EightCarsCorrupt14:
+        case RCT12TileElementType::EightCarsCorrupt15:
+            return elem_type;
+        default:
+            // Most corrupt elements were set to 0x255, but not all. Fallback to corrupt element if encountered unknown type.
+            return RCT12TileElementType::Corrupt;
```
Comment (Gymnasiast) [inline src/openrct2/rct12/RCT12.cpp:50]:
```suggestion
            return RCT12TileElementType::EightCarsCorrupt15;
```
Our old code would treat all unknown corrupt elements as the ones from 8 cars per trainer - i.e. it hid all elements after it. Our own corrupt element only hid a single element.

**Review by Gymnasiast — APPROVED** [review]

**Comment by Gymnasiast** [comment]
> Additionally: should any effort be made to see if there are consecutive corrupt elements and should they get merged to single one?

OpenRCT2 has done away with corrupt elements altogether. Instead, invisibility flags are set on following elements.

**Review by duncanspumpkin — APPROVED** [review]

**Comment by Gymnasiast** [comment]
Running into API rate limiting here, unfortunately. Let’s try again tomorrow.

**Pull request merged**